rapids-cmake/test/detail/generate_resource_spec.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_runtime_api.h>
+#include <iostream>
+#include <string>
+#include <vector>
+
+struct version {
+  int major = 1;
+  int minor = 0;
+};
+
+struct gpu {
+  gpu(int i) : id{i} {};
+  gpu(int i, const cudaDeviceProp& prop) : id{i}, memory{prop.totalGlobalMem}, slots{100} {}
+  int id        = 0;
+  size_t memory = 0;
+  int slots     = 0;
+};
+
+struct local {
+  local()
+  {
+    int nDevices = 0;
+    cudaGetDeviceCount(&nDevices);
```

Comment:
It doesn't need CUDA error handling since it is outputting a json file on standard output. When the program can't use any of the CUDA runtime API's it will generate a json output that is equal to a machine with a valid CUDA runtime but no GPUs which is what we want ( no gpu support ).